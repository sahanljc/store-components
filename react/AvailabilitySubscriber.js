export { default } from './components/AvailabilitySubscriber'
