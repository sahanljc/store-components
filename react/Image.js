import Image from './components/Image/index'

export default Image
