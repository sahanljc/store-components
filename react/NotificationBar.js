import NotificationBar from './components/NotificationBar/index'

export default NotificationBar
