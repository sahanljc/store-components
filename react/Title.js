import Title from './components/Title/index'

export default Title
