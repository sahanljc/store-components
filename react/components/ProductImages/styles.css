/**
*   Product Image CSS Module
*/

.content { }

.video { }

.image { }

.imageBlur30 { 
  filter: blur(30px);
}

.imageTransitionOpacity {
  transition: 1s;
}

.carouselGaleryCursor {
  cursor: url('./components/Carousel/cursor.svg') 8 8, default;
}

.carouselGaleryThumbs :global(.swiper-slide) {
  height: auto;
}

.carouselGaleryThumbs :global(.swiper-slide):not(:global(.swiper-slide-thumb-active)) .carouselThumbBorder {
  border-width: 0px;
}

.carouselIconCaretRight,
.carouselIconCaretLeft {
  stroke: white;
  stroke-width: 0.5px;
  -webkit-filter: drop-shadow( 0px 1px 1px #CACBCC );
  filter: drop-shadow( 0px 1px 1px #CACBCC );
}

.carouselCursorDefault:hover{
  cursor: default;
}