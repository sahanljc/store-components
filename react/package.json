{
  "private": true,
  "scripts": {
    "pretest": "yarn",
    "test": "vtex-test-tools test",
    "lint": "eslint --ext js,jsx,ts,tsx ."
  },
  "dependencies": {
    "@material-ui/core": "^3.5.1",
    "@vtex/css-handles": "^1.0.2",
    "apollo-client": "^2.5.1",
    "classnames": "^2.2.5",
    "debounce": "^1.2.0",
    "downshift": "^1.31.14",
    "graphql": "~0.13.2",
    "graphql-tag": "~2.8.0",
    "prop-types": "^15.6.2",
    "ramda": "^0.25.0",
    "react": "^16.3.2",
    "react-apollo": "~2.5.2",
    "react-content-loader": "^3.1.2",
    "react-dom": "^16.8.3",
    "react-html-parser": "^2.0.2",
    "react-id-swiper": "^1.6.8",
    "react-intl": "^2.8.0",
    "react-resize-detector": "^3.0.1",
    "react-share": "^2.1.1",
    "react-slick": "^0.23.1",
    "sanitize-html": "^1.20.0",
    "slick-carousel": "^1.8.1"
  },
  "devDependencies": {
    "@vtex/test-tools": "^0.1.2",
    "apollo-cache-inmemory": "^1.4.3",
    "eslint": "^5.16.0",
    "eslint-config-vtex-react": "^4.0.0",
    "prettier": "^1.16.4",
    "typescript": "^3.4.1"
  },
  "vtexTestTools": {
    "defaultLocale": "en"
  }
}
