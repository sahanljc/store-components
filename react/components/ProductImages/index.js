import PropTypes from 'prop-types'
import React, { useMemo, useEffect, useState } from 'react'
import debounce from 'debounce'

import Carousel from './components/Carousel'
import styles from './styles.css'

const getBestUrlIndex = thresholds => {
  const windowSize = window.innerWidth

  let bestUrlIndex = 0

  thresholds.forEach((threshold, i) => {
    if (windowSize > threshold) bestUrlIndex = i + 1
  })

  return bestUrlIndex
}

const ProductImages = props => {
  const [_, setState] = useState(0)
  const { position } = props

  const debouncedGetBestUrl = debounce(() => {
    // force update
    setState(c => c + 1)
  }, 500)

  useEffect(() => {
    window.addEventListener('resize', debouncedGetBestUrl)

    return () => {
      window.removeEventListener('resize', debouncedGetBestUrl)

      debouncedGetBestUrl.clear()
    }
  }, [debouncedGetBestUrl])

  const slides = useMemo(() => {
    const { images } = props

    if (images.length === 0) return

    return images.map(image => {
      return {
        type: 'image',
        urls: image.imageUrls,
        alt: image.imageText,
        thumbUrl: image.thumbnailUrl || image.imageUrls[0],
        bestUrlIndex: getBestUrlIndex(image.thresholds),
      }
    })
  }, [props])

  return (
    <div className={`${styles.content} w-100`}>
      <Carousel slides={slides} position={position} />
    </div>
  )
}

ProductImages.propTypes = {
  /** The position of the thumbs */
  position: PropTypes.oneOf(['left', 'right']),
  /** Array of images to be passed for the Thumbnail Slider component as a props */
  images: PropTypes.arrayOf(
    PropTypes.shape({
      /** URL of the image */
      imageUrls: PropTypes.arrayOf(PropTypes.string.isRequired),
      /** Size thresholds used to choose each image */
      thresholds: PropTypes.arrayOf(PropTypes.number),
      /** URL of the image thumbnail */
      thumbnailUrl: PropTypes.string,
      /** Text that describes the image */
      imageText: PropTypes.string.isRequired,
    })
  ),
}

ProductImages.defaultProps = {
  images: [],
  position: 'left',
}

export default ProductImages
