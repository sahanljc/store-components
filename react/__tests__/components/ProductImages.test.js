import React from 'react'
import { render } from '@vtex/test-tools/react'
import ProductImages from './../../ProductImages'

describe('<ProductImages />', () => {
  const renderComponent = customProps => {
    return render(<ProductImages {...customProps} />)
  }

  it('should be mounted', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeDefined()
  })

  it('should match the snapshot with images', () => {
    const props = {
      images: [
        {
          imageUrls: ['url'],
          thresholds: [1],
          thumbnailUrl: 'url',
          imageText: 'imageText',
        },
      ],
    }
    const { asFragment } = renderComponent(props)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot with no images', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot with thumbnails in right position', () => {
    const { asFragment } = renderComponent({ position: 'right' })
    expect(asFragment()).toMatchSnapshot()
  })
})
