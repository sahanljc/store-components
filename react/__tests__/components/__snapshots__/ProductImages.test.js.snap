// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImages /> should match the snapshot with images 1`] = `
<DocumentFragment>
  <div
    class="content w-100"
  >
    <div
      class="w-100 aspect-ratio aspect-ratio--1x1"
    >
      <svg
        class="content-loader-mock"
      >
        <rect
          height="100%"
          width="100%"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProductImages /> should match the snapshot with no images 1`] = `
<DocumentFragment>
  <div
    class="content w-100"
  >
    <div
      class="w-100 aspect-ratio aspect-ratio--1x1"
    >
      <svg
        class="content-loader-mock"
      >
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
        />
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
          y="20%"
        />
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
          y="40%"
        />
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
          y="61%"
        />
        <rect
          class="dn db-ns"
          height="89%"
          width="89%"
          x="20%"
        />
        <rect
          class="db dn-ns"
          height="100%"
          width="100%"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProductImages /> should match the snapshot with thumbnails in right position 1`] = `
<DocumentFragment>
  <div
    class="content w-100"
  >
    <div
      class="w-100 aspect-ratio aspect-ratio--1x1"
    >
      <svg
        class="content-loader-mock"
      >
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
        />
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
          y="20%"
        />
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
          y="40%"
        />
        <rect
          class="dn db-ns"
          height="18%"
          width="18%"
          y="61%"
        />
        <rect
          class="dn db-ns"
          height="89%"
          width="89%"
          x="20%"
        />
        <rect
          class="db dn-ns"
          height="100%"
          width="100%"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
